Book : update book to cover all newly added components and standardize structure of documenting crates individually (#468)
proptest-rs/proptest: book/src/intro.md

and shrinking is defined on a per-value basis instead of per-type, which
makes it more flexible and simplifies composition.

## Ecosystem

Proptest has evolved to be more than a single library. Currently the following
crates exist:

> - `proptest` - the core testing library containing traits, implementations of those
>   traits for `core`, `alloc`, `std` and some popular 3rd party crates, a test runner
>   and some sugar to make working with the test runner and traits easier
> - `proptest-derive` - a derive macro for implementing `Arbitrary` when you don't
>   have unique requirements for generating data
> - `proptest-macro` - procedural macros that make writing proptests easier
> - `proptest-state-machine` - a lightweight framework built on top of `proptest`
>   for testing systems against a reference implementation

Reviewer: adding some periods and removing use of "you", just a suggestion, feel free to ignore if you disagree :)
```suggestion
- `proptest` - the core testing library containing traits, implementations of those
  traits for `core`, `alloc`, `std` and some popular 3rd party crates, a test runner
  and some sugar to make working with the test runner and traits easier.
- `proptest-derive` - a derive macro for implementing `Arbitrary` when the applicable type does not
  have unique requirements for generating data.
- `proptest-macro` - procedural macros that make writing proptests easier.
- `proptest-state-machine` - a lightweight framework built on top of `proptest`
  for testing systems against a reference implementation.
```

## Status of the proptest ecosystem

`proptest` has been stable at a 1.x release for roughly 3 years and not seen substantial
architectural changes in quite some time -- generally receiving passive maintenance
and minor feature requests. In that time, Rust has continued evolving with the stabilization
and maturation of the async ecosystem and generic associated types (GAT). We are
evaluating what a 2.x release of proptest would look like and what features and
changes would warrant a new major version. If you have ideas, suggestions, or asks,

> please [open an issue on our github repo](https://github.com/proptest-rs/proptest/issues/new)

Reviewer: ```suggestion
please [open an issue on our github repo](https://github.com/proptest-rs/proptest/issues/new).
```
